Gemfile

```diff
@@ -1,17 +1,56 @@
-source 'https://rubygems.org'
-
-gem 'rails', '4.2.10'
-gem 'haml-rails'
-gem 'sass-rails', '~> 5.0'
-gem 'bootstrap-sass'
-gem 'uglifier', '>= 1.3.0'
-gem 'jquery-rails'
-gem 'turbolinks'
-gem 'jbuilder', '~> 2.0'
-gem 'twilio-ruby'
-
-group :production do
-  gem 'rails_12factor'
+# frozen_string_literal: true
+
+source "https://rubygems.org"
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby "2.6.5"
+
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem "rails", "~> 6.0.3", ">= 6.0.3.3"
+# Use Puma as the app server
+gem "puma", "~> 4.1"
+# Use SCSS for stylesheets
+gem "sass-rails", ">= 6"
+# Transpile app-like JavaScript. Read more: https://github.com/rails/webpacker
+gem "webpacker", "~> 4.0"
+# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
+gem "turbolinks", "~> 5"
+# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
+gem "jbuilder", "~> 2.7"
+# Use Active Model has_secure_password
+# gem 'bcrypt', '~> 3.1.7'
+
+# Reduces boot times through caching; required in config/boot.rb
+gem "bootsnap", ">= 1.4.2", require: false
+
+group :development, :test do
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem "byebug", platforms: [:mri, :mingw, :x64_mingw]
 end
 
-ruby '2.4.3'
+group :development do
+  # Access an interactive console on exception pages or by calling 'console' anywhere in the code.
+  gem "listen", "~> 3.2"
+  gem "web-console", ">= 3.3.0"
+  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
+  gem "spring"
+  gem "spring-watcher-listen", "~> 2.0.0"
+end
+
+group :test do
+  # Adds support for Capybara system testing and selenium driver
+  gem "capybara", ">= 2.15"
+  gem "selenium-webdriver"
+  # Easy installation and use of web drivers to run system tests with browsers
+  gem "webdrivers"
+end
+
+# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
+gem "tzinfo-data", platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+
+# TCS additions
+gem "haml-rails"
+
+# Mirror the tcs-rubocop version
+gem "rubocop", "0.85.0"
```

Comment:
Should we include gems like rubocop in the programming exercise? I'm not concerned with testing a candidate's knowledge of individual gems, and sometimes the point of a programming exercise is to present the candidate with code that is intentionally bad.

Reply:
My thinking with this was twofold:

1. We should have the project represent us as closely as possible, and our styleguide is part of that.
  1.1. Does that mean that the gem should be included?
  1.2. Does that mean that the code should be updated with out without the gem being included?
2. Having the code represent our style is one thing, and hopefully the candidate picks up on that and mimics it. This is part of our rubric. Having the gem and styleguide in the project without making any mention of it also acts as a breadcrumb.

As I write these points out and consider what you've said I don't really know where I stand. How does the Java exercise handle this? I'm fine with pulling that third commit out and pushing the upgraded project forward. We can always come back and add the cleanup.